### PR TITLE
FS2-2478: title and subtitle

### PR DESCRIPTION
**Pull request by JustinCCYap**
Margins are smaller to use more of the available space for the plot.

I had to remove the title from some test examples to avoid flakiness.

### CODE REVIEW

File: theSrc/test/snapshots/travis/master/testPlans/functionalTest/title_subtitle_footer-9-snap.png
Comment (chschan) [inline theSrc/test/snapshots/travis/master/testPlans/functionalTest/title_subtitle_footer-9-snap.png:1]: I don't see any snapshots showing footers?

File: theSrc/scripts/LabeledScatter.js [hunk 1]
```diff
@@ -414,6 +443,90 @@ class LabeledScatter {
     return annotations
   }
 
+  adjustTitles (plotly_chart_layout, config) {
```
Comment (chschan) [inline theSrc/scripts/LabeledScatter.js:446]: Why does adjustTitles have to be called in the `after_plot` event handler? I would have thought that zooming or clicking on things wouldn't affect the adjusted titles?

File: theSrc/scripts/LabeledScatter.js [hunk 2]
```diff
@@ -414,6 +443,90 @@ class LabeledScatter {
...
+    if (index === -1) {
+      return null
+    }
+    const annotations = d3.select(this.rootElement).selectAll('.annotation')
+    return annotations[0][0] ? d3.select(annotations[0][index]) : null
+  }
+
+  titleBottom (config) {
+    if (config.title && config.title.length > 0) {
+      return config.title.split('<br>').length * config.titleFontSize * LINE_HEIGHT_AS_PROPORTION_OF_FONT_SIZE
```
Comment (chschan) [inline theSrc/scripts/LabeledScatter.js]: This seems to be the only place you use `LINE_HEIGHT_AS_PROPORTION_OF_FONT_SIZE` in LabeledScatter.js? Maybe these functions should go in to PlotlyChartElements.js and you just export them across?

Comment (JustinCCYap) [reply theSrc/scripts/LabeledScatter.js]: Yes, I've moved `titleBottom` to PlotlyChartElements.js in 112c17f24bfb8460afe6351f6fc3f3c2a58a9843

File: theSrc/test/snapshots/travis/master/testPlans/functionalTest/title_subtitle_footer-9-snap.png
Comment (JustinCCYap) [reply theSrc/test/snapshots/travis/master/testPlans/functionalTest/title_subtitle_footer-9-snap.png:1]: These tests were inherited from rhtmlLabeledScatter. Footers will be shown in https://numbers.atlassian.net/browse/FS2-2491

File: theSrc/scripts/LabeledScatter.js (hunk 1 above)
Comment (JustinCCYap) [reply theSrc/scripts/LabeledScatter.js:446]: Plotly moves the title and annotation back to its original location after zooming, so we need to readjust the titles.

File: theSrc/scripts/LabeledScatter.js (hunk 2 above)
Comment (chschan) [reply theSrc/scripts/LabeledScatter.js]: Maybe the function name should be `titleHeight` or `titleVerticalSpace` instead?

**Review by chschan — APPROVED** [review]

**Pull request merged**